google-cloud-spanner/src/main/java/com/google/cloud/spanner/Type.java

```diff
@@ -287,6 +296,7 @@ public enum Code {
     FLOAT64(TypeCode.FLOAT64, "double precision"),
     FLOAT32(TypeCode.FLOAT32, "real"),
     STRING(TypeCode.STRING, "character varying"),
+    UUID(TypeCode.STRING, "uuid"),
```

Comment:
As per design doc a new TypeCode `UUID` will be added. This will recognize the input serialized string value as UUID in backend to store it appropriately.
```suggestion
    UUID(TypeCode.UUID, "uuid"),
```